sqlglot/dataframe/sql/dataframe.py

```diff
@@ -365,20 +365,27 @@ def select(self, *cols, **kwargs) -> DataFrame:
                     x.this for x in get_tables_from_expression_with_join(self.expression)
                 ]
                 cte_names_in_join = [x.this for x in join_table_identifiers]
+                # If we a columns resolves to multiple CTE expressions then we want to use each CTE left-to-right
```

Comment:
if we have

Reply:
or if a

Reply:
why do you want to use ctes left to right?

Reply:
To match Spark behavior